### PR TITLE
Add calendars/index.yaml file: metadata for the index list

### PR DESCRIPTION
**Pull request by rkdarst**
- A file named index.yaml can be provided, which is not a calendar but
  contains metadata that can be used in the index.html template file.
  So far, this can be used to set the title and add a description
  above the calendar list.  This is useful to make the page look a bit
  more professional.
- The file is currently hardcoded to the name index.yaml, which I
  guess is OK since index is commonly used for indexes.
- A bit more questionable is that it is in calendars/index.yaml (as
  in, mixed in with the other calendars).  Perhaps this makes sense if
  this is seen as "data".  The practial use is that other consumers
  don't need to change their build.sh to add in new command line
  arguments.
  - Basically, when reading the calendar files, if something is named
    `index.*`, treat this as the metadata.
- Add a sample index.yaml file.
- Review: is there anything very off here?  Anyway, it can be iterated

### CODE REVIEW

**Comment by bast** [comment]
I also thought that having yml-based metadata configuration is a good idea but mixing it under calendars/ does not sound good to me. But I also don't know where else to place it.

Perhaps I would have data/ folder with customization yml file and beneath calendars/?

**Comment by bast** [comment]
Like this?
```
.
└── data
    ├── calendars
    │   ├── team.yml
    │   └── workshops.yml
    └── index.yml
```

**Comment by rkdarst** [comment]
What about `_config.yaml` in with calendars?

Somehow, I'm not too worried about mixing the config with the content - it happens in many other projects, such as jekyll.  Then, either the layout becomes fixed (right now it's general globbing), or a new argument is needed for "here is the config".

Thinking about it, I am inclined to rename it to `_config.yaml` and then leaving it in the calendars directory.  The location can be iterated someday when the overall directory layout is re-thought - or a new `-c` option can be added.
